deployment/azure/create_and_deploy_certificates.sh

```diff
@@ -0,0 +1,169 @@
+	sed -i "s/<NAME_OF_SECRET_OF_SERVICE_PRINCIPAL_WITH_ACCESS_TO_AZURE_DNS>/$NAME_OF_SECRET_OF_SERVICE_PRINCIPAL_WITH_ACCESS_TO_AZURE_DNS/" $CUSTOM_RESOURCE
+	sed -i "s/<DNS_ZONE_NAME>/$DNS_ZONE_NAME/" $CUSTOM_RESOURCE
+}
+
+function createCertificateDescriptor() {
+	local SIMPLE_NAME=$1
+	local NAMESPACE=$2
+	local TARGET_FILE=$SIMPLE_NAME-certificate.yaml
+	local TEMPLATE_FILE=certificate.yaml
+	local RESOURCE_TEMPLATE=$CERT_TEMPLATES/$TEMPLATE_FILE
+	local CUSTOM_RESOURCE=$CERT_RESOURCES/$TARGET_FILE
+	
+	echo "Creating resource descriptor $CUSTOM_RESOURCE ..."
+	cp $RESOURCE_TEMPLATE $CUSTOM_RESOURCE
+	
+	sed -i "s/<SIMPLE_NAME>/$SIMPLE_NAME/" $CUSTOM_RESOURCE
+	sed -i "s/<CLUSTER_NAME>/$CLUSTER_NAME/" $CUSTOM_RESOURCE
+	sed -i "s/<DNS_ZONE_NAME>/$DNS_ZONE_NAME/" $CUSTOM_RESOURCE
+	sed -i "s/<NAMESPACE>/$NAMESPACE/" $CUSTOM_RESOURCE
+	sed -i "s/<NAME_OF_CLUSTER_ISSUER>/$NAME_OF_CLUSTER_ISSUER/" $CUSTOM_RESOURCE
+}
+
+echo
+echo "####################################################################################################"
+echo "####################################################################################################"
+echo "###############     Install Cert-Manager and Get Certificates from Let's Encrypt     ###############"
+echo "####################################################################################################"
+echo "####################################################################################################"
+echo
+echo "########## Initialization ##########"
+
+RESOURCE_TEMPLATES=$SCRIPTPATH/resource-descriptor-templates
+CLUSTER_ISSUER_TEMPLATES=$RESOURCE_TEMPLATES/cluster-issuers
+CERT_TEMPLATES=$RESOURCE_TEMPLATES/certificates
+
+RESOURCE_DESCRIPTORS=$SCRIPTPATH/resource-descriptors
+CLUSTER_ISSUER_RESOURCES=$RESOURCE_DESCRIPTORS/cluster-issuers
+CERT_RESOURCES=$RESOURCE_DESCRIPTORS/certificates
+
+echo "### Create resource descriptors ###"
+mkdir -p $CLUSTER_ISSUER_RESOURCES
+customizeClusterIssuer "letsencrypt-prod-clusterissuer-dns01.yaml"
+customizeClusterIssuer "letsencrypt-staging-clusterissuer-dns01.yaml"
+
+mkdir -p $CERT_RESOURCES
+createCertificateDescriptor "gateway" $NAMESPACE_TO_STORE_CERTIFICATES_IN
+
+echo "### Install cert-manager ###"
+INSTALLED_CERT_MANAGER_CHART_VERSION="$(kubectl get deployments -n cert-manager cert-manager -o yaml --ignore-not-found=true \
+	| yq .metadata.labels.chart \
+	| tr -d '\"' \
+	| sed -e 's/cert-manager-v//')"
+
+if [[ "$INSTALLED_CERT_MANAGER_CHART_VERSION" == "$CERT_MANAGER_CHART_VERSION" ]]; then
+	echo "cert-manager is up-to-date"
+elif [[ "" != "$INSTALLED_CERT_MANAGER_CHART_VERSION" ]]; then
+	# See
+	# * https://helm.sh/docs/using_helm/#installing-helm
+	# * https://docs.helm.sh/using_helm/#securing-your-helm-installation
+	# * https://docs.cert-manager.io/en/latest/tasks/upgrading/index.html
+	# Note that tiller is started, sent to the background and eventually terminated
+	echo "Upgrading cert-manager using local tiller ..."
```

Comment:
Can me mention this and that Helm is required in a more prominent phase like the Readme files.